library/Fission/CLI/Environment.hs

```diff
@@ -94,29 +94,30 @@ removeConfigFile = do
 
 -- | Retrieves a Fission Peer from local config
 --   If not found we retrive from the network and store
-getOrRetrievePeer ::
+getOrRetrievePeers ::
   ( MonadUnliftIO  m
   , MonadLogger    m
   , MonadWebClient m
   )
   => Environment
-  -> m (Maybe IPFS.Peer)
-getOrRetrievePeer Environment {peers = (peer : _)} = do
-  logDebug @Text "Retrieved Peer from .fission.yaml"
-  return $ Just peer
-
-getOrRetrievePeer Environment {peers = []} =
+  -> m [IPFS.Peer]
+getOrRetrievePeers Environment {peers = []} =
   Peers.getPeers >>= \case
     Left err -> do
       logError $ displayShow err
       logDebug @Text "Unable to retrieve peers from the network"
-      return Nothing
+      return []
 
     Right nonEmptyPeers -> do
-      logDebug @Text "Retrieved Peer from API"
       path <- globalEnv
-      Override.writeMerge path $ mempty { peers = NonEmpty.toList nonEmptyPeers }
-      return . Just $ NonEmpty.head nonEmptyPeers
+      let peers = NonEmpty.toList nonEmptyPeers
+      logDebug $ "Retrieved Peers from API, and writing to ~/.fission.yaml: " <> textShow peers
+      Override.writeMerge path $ mempty { peers }
+      return peers
+
+getOrRetrievePeers Environment {peers = peers} = do
```

Comment:
```haskell
getOrRetrievePeers Environment {peers = peers}
```

Can be written without the `=`

```haskell
getOrRetrievePeers Environment {peers}
```

or even as a wildcard:

```haskell
getOrRetrievePeers Environment {..}
```

Reply:
Resolved?

Reply:
I.... I swear I fixed this. like have a distinct memory of doing it. spooky :ghost: 
ah well thanks for double checking me :sweat_smile: